src/models/types.ts

```diff
@@ -16,3 +18,24 @@ export interface TestStage {
   name: string;
   command: string;
 }
+
+export interface ProgressDetails {
+  current_step: number;
+}
+
+export interface ProgressResponse {
+  id: string;
+  user_id: string;
+  challenge_id: string;
+  status: string;
+  progress_details: ProgressDetails;
+  created_at: string;
+  updated_at: string;
+  total_stages: number;
```

Comment:
This is not included in the Progress object from the backend
```suggestion
```

Reply:
You are right, I am meant to remove this.